From 71a37282361ba54a3a06ad66b939fb28ea717b32 Mon Sep 17 00:00:00 2001
From: Luc Verhaegen <libv@codethink.co.uk>
Date: Sun, 28 Oct 2012 12:26:36 +0100
Subject: [PATCH 878/944] video:sunxi:disp: clean up messaging macros

Well, rip out their guts and replace them with something sane
and actually working.

Signed-off-by: Luc Verhaegen <libv@codethink.co.uk>
---
 drivers/video/sunxi/disp/bsp_display.h    |  15 +---
 drivers/video/sunxi/disp/de_be.c          |   4 +-
 drivers/video/sunxi/disp/dev_disp.c       |   2 +-
 drivers/video/sunxi/disp/dev_fb.c         |   2 +-
 drivers/video/sunxi/disp/disp_display.c   |   6 +-
 drivers/video/sunxi/disp/disp_display_i.h |   2 +-
 drivers/video/sunxi/disp/disp_lcd.c       | 118 +++++++++++++++---------------
 7 files changed, 67 insertions(+), 82 deletions(-)

diff --git a/drivers/video/sunxi/disp/bsp_display.h b/drivers/video/sunxi/disp/bsp_display.h
index 5bc7d1b..cc18394 100644
--- a/drivers/video/sunxi/disp/bsp_display.h
+++ b/drivers/video/sunxi/disp/bsp_display.h
@@ -57,19 +57,8 @@ typedef unsigned int __hdle;
 
 #include <linux/drv_display.h>
 
-#if 1
-#define OSAL_PRINTF(msg...) {printk(KERN_WARNING "[DISP] ");printk(msg);}
-#define __inf(msg...)
-#define __msg(msg...)
-#define __wrn(msg...)       {printk(KERN_WARNING "[DISP WRN] file:%s,line:%d:    ",__FILE__,__LINE__);printk(msg);}
-#define __here__
-#else
-#define OSAL_PRINTF(msg...) {printk(KERN_WARNING "[DISP] ");printk(msg);}
-#define __inf(msg...)       {printk(KERN_WARNING "[DISP] ");printk(msg);}
-#define __msg(msg...)       {printk(KERN_WARNING "[DISP] file:%s,line:%d:    ",__FILE__,__LINE__);printk(msg);}
-#define __wrn(msg...)       {printk(KERN_WARNING "[DISP WRN] file:%s,line:%d:    ",__FILE__,__LINE__);printk(msg);}
-#define __here__            {printk(KERN_WARNING "[DISP] file:%s,line:%d\n",__FILE__,__LINE__);}
-#endif
+#define __inf(msg, ...) pr_debug("[DISP] " msg, ##__VA_ARGS__)
+#define __wrn(msg, ...) pr_warn("[DISP] " msg, ##__VA_ARGS__)
 
 typedef struct {
 	__u32 base_image0;
diff --git a/drivers/video/sunxi/disp/de_be.c b/drivers/video/sunxi/disp/de_be.c
index 56fbe9a..eaee97a 100644
--- a/drivers/video/sunxi/disp/de_be.c
+++ b/drivers/video/sunxi/disp/de_be.c
@@ -1381,8 +1381,8 @@ __s32 DE_BE_Set_Enhance(__u8 sel, __u32 out_csc, __u32 out_color_range,
 		DE_BE_WUINT32(sel, DE_BE_OUT_COLOR_B_COEFF_OFF + 4 * i,
 			      *(pt + 8 + i));
 #if 0
-		OSAL_PRINTF("R:%x,\tG:%x,\tB:%x\n", *(pt + i), *(pt + 4 + i),
-			    *(pt + 8 + i));
+		DE_WRN("R:%x,\tG:%x,\tB:%x\n", *(pt + i), *(pt + 4 + i),
+		       *(pt + 8 + i));
 #endif
 	}
 
diff --git a/drivers/video/sunxi/disp/dev_disp.c b/drivers/video/sunxi/disp/dev_disp.c
index e7c3ed1..c1b8b7b 100644
--- a/drivers/video/sunxi/disp/dev_disp.c
+++ b/drivers/video/sunxi/disp/dev_disp.c
@@ -646,7 +646,7 @@ long disp_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 	    cmd != DISP_CMD_GET_OUTPUT_TYPE && cmd != DISP_CMD_SCN_GET_WIDTH &&
 	    cmd != DISP_CMD_SCN_GET_HEIGHT && cmd != DISP_CMD_VIDEO_SET_FB &&
 	    cmd != DISP_CMD_VIDEO_GET_FRAME_ID) {
-		OSAL_PRINTF("cmd:0x%x,%ld,%ld\n", cmd, ubuffer[0], ubuffer[1]);
+		DE_WRN("cmd:0x%x,%ld,%ld\n", cmd, ubuffer[0], ubuffer[1]);
 	}
 #endif
 
diff --git a/drivers/video/sunxi/disp/dev_fb.c b/drivers/video/sunxi/disp/dev_fb.c
index 87bdf12..3eb778c 100644
--- a/drivers/video/sunxi/disp/dev_fb.c
+++ b/drivers/video/sunxi/disp/dev_fb.c
@@ -309,7 +309,7 @@ fb_draw_gray_pictures(__u32 base, __u32 width, __u32 height,
 				sys_put_wvalue(addr, value);
 			}
 		}
-		OSAL_PRINTF("----%d\n", time * 15);
+		DE_WRN("----%d\n", time * 15);
 		msleep(1000 * 5);
 	}
 	return 0;
diff --git a/drivers/video/sunxi/disp/disp_display.c b/drivers/video/sunxi/disp/disp_display.c
index 401411c..ecf9ae7 100644
--- a/drivers/video/sunxi/disp/disp_display.c
+++ b/drivers/video/sunxi/disp/disp_display.c
@@ -266,7 +266,7 @@ __s32 BSP_disp_print_reg(__bool b_force_on, __u32 id)
 	}
 
 	if (b_force_on) {
-		OSAL_PRINTF("%s", str);
+		DE_WRN("%s", str);
 	} else {
 		DE_INF("%s", str);
 	}
@@ -280,8 +280,8 @@ __s32 BSP_disp_print_reg(__bool b_force_on, __u32 id)
 		reg[3] = sys_get_wvalue(base + i + 12);
 
 		if (b_force_on) {
-			OSAL_PRINTF("0x%08x:%08x,%08x:%08x,%08x\n", base + i,
-				    reg[0], reg[1], reg[2], reg[3]);
+			DE_WRN("0x%08x:%08x,%08x:%08x,%08x\n", base + i,
+			       reg[0], reg[1], reg[2], reg[3]);
 		} else {
 			DE_INF("0x%08x:%08x,%08x:%08x,%08x\n", base + i, reg[0],
 			       reg[1], reg[2], reg[3]);
diff --git a/drivers/video/sunxi/disp/disp_display_i.h b/drivers/video/sunxi/disp/disp_display_i.h
index afbd088..985b644 100644
--- a/drivers/video/sunxi/disp/disp_display_i.h
+++ b/drivers/video/sunxi/disp/disp_display_i.h
@@ -26,8 +26,8 @@
 #include "ebios_lcdc_tve.h"
 
 #define DE_INF __inf
-#define DE_MSG __msg
 #define DE_WRN __wrn
+
 #define OSAL_IRQ_RETURN IRQ_HANDLED
 
 #define HANDTOID(handle)  ((handle) - 100)
diff --git a/drivers/video/sunxi/disp/disp_lcd.c b/drivers/video/sunxi/disp/disp_lcd.c
index bbf6842..56461fa 100644
--- a/drivers/video/sunxi/disp/disp_lcd.c
+++ b/drivers/video/sunxi/disp/disp_lcd.c
@@ -134,98 +134,95 @@ Lcd_Panel_Parameter_Check(__u32 sel)
 		if (Lcd_Panel_Err_Flag != 0) {
 			__u32 i;
 			for (i = 0; i < 200; i++) {
-				OSAL_PRINTF("*** Lcd in danger...\n");
+				DE_WRN("*** Lcd in danger...\n");
 			}
 		}
 
-		OSAL_PRINTF("*************************************************"
-			    "****************\n");
-		OSAL_PRINTF("***\n");
-		OSAL_PRINTF("*** LCD Panel Parameter Check\n");
-		OSAL_PRINTF("***\n");
-		OSAL_PRINTF("***             by dulianping\n");
-		OSAL_PRINTF("***\n");
-		OSAL_PRINTF("*************************************************"
-			    "****************\n");
-
-		OSAL_PRINTF("***\n");
-		OSAL_PRINTF("*** Interface:");
+		DE_WRN("********************************************************"
+		       "*********\n");
+		DE_WRN("***\n");
+		DE_WRN("*** LCD Panel Parameter Check\n");
+		DE_WRN("***\n");
+		DE_WRN("***             by dulianping\n");
+		DE_WRN("***\n");
+		DE_WRN("********************************************************"
+		       "*********\n");
+
+		DE_WRN("***\n");
+		DE_WRN("*** Interface:");
 		if (info->lcd_if == 0 && info->lcd_hv_if == 0) {
-			OSAL_PRINTF("*** Parallel HV Panel\n");
+			DE_WRN("*** Parallel HV Panel\n");
 		} else if (info->lcd_if == 0 && info->lcd_hv_if == 1) {
-			OSAL_PRINTF("*** Serial HV Panel\n");
+			DE_WRN("*** Serial HV Panel\n");
 		} else if (info->lcd_if == 0 && info->lcd_hv_if == 2) {
-			OSAL_PRINTF("*** Serial YUV Panel\n");
+			DE_WRN("*** Serial YUV Panel\n");
 		} else if (info->lcd_if == 3 && info->lcd_lvds_bitwidth == 0) {
-			OSAL_PRINTF("*** 24Bit LVDS Panel\n");
+			DE_WRN("*** 24Bit LVDS Panel\n");
 		} else if (info->lcd_if == 3 && info->lcd_lvds_bitwidth == 1) {
-			OSAL_PRINTF("*** 18Bit LVDS Panel\n");
+			DE_WRN("*** 18Bit LVDS Panel\n");
 		} else if (info->lcd_if == 1 && info->lcd_cpu_if == 0) {
-			OSAL_PRINTF("*** 18Bit CPU Panel\n");
+			DE_WRN("*** 18Bit CPU Panel\n");
 		} else if (info->lcd_if == 1 && info->lcd_cpu_if == 4) {
-			OSAL_PRINTF("*** 16Bit CPU Panel\n");
+			DE_WRN("*** 16Bit CPU Panel\n");
 		} else {
-			OSAL_PRINTF("\n");
-			OSAL_PRINTF("*** lcd_if:     %d\n", info->lcd_if);
-			OSAL_PRINTF("*** lcd_hv_if:  %d\n", info->lcd_hv_if);
-			OSAL_PRINTF("*** lcd_cpu_if: %d\n", info->lcd_cpu_if);
+			DE_WRN("\n");
+			DE_WRN("*** lcd_if:     %d\n", info->lcd_if);
+			DE_WRN("*** lcd_hv_if:  %d\n", info->lcd_hv_if);
+			DE_WRN("*** lcd_cpu_if: %d\n", info->lcd_cpu_if);
 		}
 
 		if (info->lcd_frm == 0) {
-			OSAL_PRINTF("*** Lcd Frm Disable\n");
+			DE_WRN("*** Lcd Frm Disable\n");
 		} else if (info->lcd_frm == 1) {
-			OSAL_PRINTF("*** Lcd Frm to RGB666\n");
+			DE_WRN("*** Lcd Frm to RGB666\n");
 		} else if (info->lcd_frm == 2) {
-			OSAL_PRINTF("*** Lcd Frm to RGB565\n");
+			DE_WRN("*** Lcd Frm to RGB565\n");
 		}
 
-		OSAL_PRINTF("***\n");
-		OSAL_PRINTF("*** Timing:\n");
-		OSAL_PRINTF("*** lcd_x:      %d\n", info->lcd_x);
-		OSAL_PRINTF("*** lcd_y:      %d\n", info->lcd_y);
-		OSAL_PRINTF("*** lcd_ht:     %d\n", info->lcd_ht);
-		OSAL_PRINTF("*** lcd_hbp:    %d\n", info->lcd_hbp);
-		OSAL_PRINTF("*** lcd_vt:     %d\n", info->lcd_vt);
-		OSAL_PRINTF("*** lcd_vbp:    %d\n", info->lcd_vbp);
-		OSAL_PRINTF("*** lcd_hspw:   %d\n", info->lcd_hv_hspw);
-		OSAL_PRINTF("*** lcd_vspw:   %d\n", info->lcd_hv_vspw);
-		OSAL_PRINTF("*** lcd_frame_frq:  %dHz\n", lcd_fclk_frq);
+		DE_WRN("***\n");
+		DE_WRN("*** Timing:\n");
+		DE_WRN("*** lcd_x:      %d\n", info->lcd_x);
+		DE_WRN("*** lcd_y:      %d\n", info->lcd_y);
+		DE_WRN("*** lcd_ht:     %d\n", info->lcd_ht);
+		DE_WRN("*** lcd_hbp:    %d\n", info->lcd_hbp);
+		DE_WRN("*** lcd_vt:     %d\n", info->lcd_vt);
+		DE_WRN("*** lcd_vbp:    %d\n", info->lcd_vbp);
+		DE_WRN("*** lcd_hspw:   %d\n", info->lcd_hv_hspw);
+		DE_WRN("*** lcd_vspw:   %d\n", info->lcd_hv_vspw);
+		DE_WRN("*** lcd_frame_frq:  %dHz\n", lcd_fclk_frq);
 
 		/* Print Error */
-		OSAL_PRINTF("***\n");
+		DE_WRN("***\n");
 		if (Lcd_Panel_Err_Flag & BIT0) {
-			OSAL_PRINTF("*** Err01: Violate \"lcd_hbp > "
-				    "lcd_hspw\"\n");
+			DE_WRN("*** Err01: Violate \"lcd_hbp > lcd_hspw\"\n");
 		}
 		if (Lcd_Panel_Err_Flag & BIT1) {
-			OSAL_PRINTF("*** Err02: Violate \"lcd_vbp > "
-				    "lcd_vspw\"\n");
+			DE_WRN("*** Err02: Violate \"lcd_vbp > lcd_vspw\"\n");
 		}
 		if (Lcd_Panel_Err_Flag & BIT2) {
-			OSAL_PRINTF("*** Err03: Violate \"lcd_ht >= "
-				    "(lcd_hbp+lcd_x*%d+4)\"\n", cycle_num);
+			DE_WRN("*** Err03: Violate \"lcd_ht >= "
+			       "(lcd_hbp+lcd_x*%d+4)\"\n", cycle_num);
 		}
 		if (Lcd_Panel_Err_Flag & BIT3) {
-			OSAL_PRINTF("*** Err04: Violate \"(lcd_vt/2) >= "
-				    "(lcd_vbp+lcd_y+2)\"\n");
+			DE_WRN("*** Err04: Violate \"(lcd_vt/2) >= "
+			       "(lcd_vbp+lcd_y+2)\"\n");
 		}
 		if (Lcd_Panel_Err_Flag & BIT10) {
-			OSAL_PRINTF("*** Err10: Violate \"lcd_io_cfg0\", "
-				    "use \"0x00000000\" or \"0x04000000\"");
+			DE_WRN("*** Err10: Violate \"lcd_io_cfg0\", "
+			       "use \"0x00000000\" or \"0x04000000\"");
 		}
 		if (Lcd_Panel_Wrn_Flag & BIT0) {
-			OSAL_PRINTF("*** WRN01: Recommend \"lcd_frm = 1\"\n");
+			DE_WRN("*** WRN01: Recommend \"lcd_frm = 1\"\n");
 		}
 		if (Lcd_Panel_Wrn_Flag & BIT1) {
-			OSAL_PRINTF("*** WRN02: Recommend \"lcd_frm = 2\"\n");
+			DE_WRN("*** WRN02: Recommend \"lcd_frm = 2\"\n");
 		}
 		if (Lcd_Panel_Wrn_Flag & BIT2) {
-			OSAL_PRINTF("*** WRN03: Recommend \"lcd_dclk_frq = "
-				    "%d\"\n",
-				    ((info->lcd_vt / 2) * info->lcd_ht) *
-				    60 / (1000 * 1000));
+			DE_WRN("*** WRN03: Recommend \"lcd_dclk_frq = %d\"\n",
+			       ((info->lcd_vt / 2) * info->lcd_ht) * 60 /
+			       (1000 * 1000));
 		}
-		OSAL_PRINTF("***\n");
+		DE_WRN("***\n");
 
 		if (Lcd_Panel_Err_Flag != 0) {
 			__u32 image_base_addr;
@@ -247,12 +244,11 @@ Lcd_Panel_Parameter_Check(__u32 sel)
 			/* open layer */
 			sys_put_wvalue(image_base_addr + 0x800, reg_value);
 
-			OSAL_PRINTF("*** Try new parameters,you can make it "
-				    "pass!\n");
+			DE_WRN("*** Try new parameters,you can make it pass!\n");
 		}
-		OSAL_PRINTF("*** LCD Panel Parameter Check End\n");
-		OSAL_PRINTF("*************************************************"
-			    "****************\n");
+		DE_WRN("*** LCD Panel Parameter Check End\n");
+		DE_WRN("********************************************************"
+		       "*********\n");
 	}
 }
 
-- 
1.8.0

